core/src/test/java/com/graphhopper/routing/lm/LMApproximatorTest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.graphhopper.routing.lm;
+
+import com.graphhopper.Repeat;
+import com.graphhopper.RepeatRule;
+import com.graphhopper.routing.Dijkstra;
+import com.graphhopper.routing.Path;
+import com.graphhopper.routing.util.CarFlagEncoder;
+import com.graphhopper.routing.util.DefaultEdgeFilter;
+import com.graphhopper.routing.util.EncodingManager;
+import com.graphhopper.routing.util.TraversalMode;
+import com.graphhopper.routing.weighting.*;
+import com.graphhopper.storage.Directory;
+import com.graphhopper.storage.GraphBuilder;
+import com.graphhopper.storage.GraphHopperStorage;
+import com.graphhopper.storage.RAMDirectory;
+import com.graphhopper.util.EdgeIterator;
+import com.graphhopper.util.GHUtility;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+
+public class LMApproximatorTest {
+
+    @Rule
+    public RepeatRule repeatRule = new RepeatRule();
+
+    @Before
+    public void init() {
+    }
+
+    @Test
+    @Repeat(times = 5)
+    public void randomGraph() {
+        final long seed = System.nanoTime();
+        System.out.println("random Graph seed: " + seed);
+        run(seed);
+    }
+
+    @Test
+    public void fail7() {
+        run(337586249813919L);
```

Comment:
I'd rather not have these tests depend on a specific seed, because this way they are strictly coupled to the exact usage of the random number generator in `GHUtiliy#randomGraph`. The lazy method would be to duplicate the randomGraph method in this class and never change it again. The better method would be writing down the corresponding graph explicitly (there is GHUtiliy#printGraphForUnitTest to do this quickly).

Reply:
Thanks, I agree. The even lazier and almost-as-good method would be to remove maybe all but one of them completely. They're reasonably well covered by the general random test, I think.